linux/jdk/redhat/src/main/packaging/build.sh

```diff
@@ -4,14 +4,18 @@ set -euxo pipefail
 # Ensure necessary directories for rpmbuild operation are present.
 rpmdev-setuptree
 
-# Build all spec files we can find
-targets="x86_64 ppc64le s390x aarch64 armv7hl"
+# Build specified target or build all (not s390x on jdk8)
+if ${buildArch} != "all"; then
+	targets=${buildArch}
+elif ${buildVersion} != "8"; then
+  	targets="x86_64 ppc64le aarch64 armv7hl s390x"
+else
+	targets="x86_64 ppc64le aarch64 armv7hl"	
+fi
+
 for spec in "$(ls /home/builder/build/generated/packaging/*.spec)"; do
 	spectool -g -R "$spec";
 	rpmbuild --nodeps -bs "$spec";
-	if [[ "$spec" =~ "temurin-8-jdk" ]]; then
-		targets="x86_64 ppc64le aarch64 armv7hl"
-	fi
 	for target in $targets; do
```

Comment:
A exception for targets=source should be considered?

Reply:
yes, 'source' should be added

Reply:
I looked up the change, in the jenkins dropdown i have this source, but the code it does not need source as input
because i does the` rpmbuild -bs `first then do the real `rpmbuild -target `
so we actually do not need an option in jenkins to choose for `source`